src/pspm_bf_spsrf_box.m

```diff
@@ -1,60 +1,64 @@
 function [bs, x] = pspm_bf_spsrf_box(varargin)
 % pspm_bf_spsrf_box constructs a boxcar function to produce the averaged
-% scanpath speed over a 2-s time window in the end of SOA, which equals 
+% scanpath speed over a 2-s time window in the end of SOA, which equals
 % to scanpath length over a 2-s time window divided by 2/s
 %
-% FORMAT: [bs, x] = pspm_bf_spsrf_box(td, soa) 
+% FORMAT: [bs, x] = pspm_bf_spsrf_box(td, soa)
 %     OR: [bs, x] = pspm_bf_spsrf_box([td, soa])
 %
 % FORMAT: [bf p] = pspm_bf_spsrf_box(td, soa)
 %         with  td: time resolution in s
 %
 % REFERENCE
-% (1) Xia Y, Melinscak F, Bach DR (2020) 
+% (1) Xia Y, Melinscak F, Bach DR (2020)
 % Saccadic Scanpath Length: An Index for Human Threat Conditioning
 % Behavioral Research Methods (submitted)
 %________________________________________________________________________
 % PsPM 4.0
 
 % initialize
 global settings
-if isempty(settings), pspm_init; end;
+if isempty(settings), pspm_init; end
 
 % check input arguments
-if nargin==0
-    errmsg='No sampling interval stated'; warning('ID:invalid_input', errmsg); return;
+if nargin == 0
+  errmsg='No sampling interval stated'; warning('ID:invalid_input', errmsg); return;
 elseif nargin == 1
-    n_el = numel(varargin{1});
-    td = varargin{1}(1);
-    if n_el > 1, soa = varargin{1}(2); else , soa=3.5; end;
+  n_el = numel(varargin{1});
+  td = varargin{1}(1);
+  if n_el > 1
+    soa = varargin{1}(2); 
+  else
+    soa = settings.glm(10).filter.soa;

```

Comment:
no need to change like this. the values are the input arguments (varargin) when calling this function. Just keep it as it was, so the default SOA will be 3.5. 

when we want to do model inversion (i.e. run glm), we first set model information.
datafile = .... (filename)
timing = ...(timing for glm);
soa = ...(experimental soa, if not set, the function should take it as default, i.e. 3.5 s here);
mod = 'sps';
outputfile = ... (the output filename after glm);

% option 1: now set the model for glm without specific SOA, but just take the default RF and SOA
model.timing = timing;
model.modelspec = mod;
model.modelfile = outputfile;
model.datafile = datafile;
model.timeunits = 'markers'; %(i.e. timing according to trial markers)
options = struct('overwrite', 1);
mdl = pspm_glm(model, options);

the codes then go to pspm_glm, there in the function, line 348 % check & get basis function
it will check whether bf is pre-set or not. Here you can see that it takes settings.glm(modno).cbf;
and then later in line 373, it uses model.bf.args(:) as input argument for this bf function (i.e. pspm_bf_spsrf_box)
so please do not change anything in this bf function, only change the pspm_init is enough, as I suggested.

% option 2: now set the model for glm with specific SOA,
everything of model is the same as previous. But add the following:
model.bf.fhandle = @pspm_bf_spsrf_box;
model.bf.args = inarg.soa;

mdl = pspm_glm(model, options);

so now when the code runs into pspm_glm, it will take the soa I specifically set.

Conclusion:
if you want to change in your way, you need to change a bunch of codes and scripts. Could you please just simply change cbf settings of sps modality in pspm_init ?

Reply:
Hi Yanfang, thank you for the very explicit explanation. Could I ask you to check the newly implemented solution [here](https://github.com/bachlab/PsPM/pull/307#issue-1000513029) and let me know if you are happy with it? Thanks.